app/lib/actions/katello/repository/create.rb

```diff
@@ -33,6 +34,12 @@ def plan(repository, args = {})
 
             concurrence do
               plan_self(:repository_id => repository.id, :clone => clone)
+              repository.product.alternate_content_sources.with_type(repository.content_type).each do |acs|
+                acs.smart_proxies.each do |smart_proxy|
+                  smart_proxy_acs = ::Katello::SmartProxyAlternateContentSource.create(alternate_content_source_id: acs.id, smart_proxy_id: smart_proxy.id, repository_id: repository.id)
+                  plan_action(Pulp3::Orchestration::AlternateContentSource::Create, smart_proxy_acs)
+                end
+              end
             end
           end
         end
```

Comment:
This doesn't behave as intended if the ACS was created with an empty product and repos added later to it.

Reply:
I see why this is happening. With a product that has no repositories, no smart_proxy_alternate_content_sources get created. So, `acs.smart_proxies` returns `[]`.